extends Node2D

const battleunits = preload("res://BattleUnits.tres")

#zona de declaraci칩n de variables
onready var botones = $UI/buttonpanel/GridContainer

#funci칩n para controlar el turno del jugador, el yield
#espera hasta que el player emita la se침al de turnend
func turnoplayer():
	var player = battleunits.player
	botones.show()
	player.acc = player.maxacc
	yield(player,"turnEnd")
	turnoenemigo()

#funci칩n de turno enemigo, para evitar que el player
#toque cosas que no debe se ocultan los botones, luego
#se llama a la func atacar del enemigo
func turnoenemigo():
	botones.hide()
	var enemy = battleunits.enemy
	if enemy != null:
		enemy.atacar()
		turnoplayer()


func _ready():
	var enemy = battleunits.enemy
	var player = battleunits.player
	if player.vel > enemy.vel:
		turnoplayer()
	else:
		turnoenemigo()

func _on_enemy_muerto():
	botones.hide()
